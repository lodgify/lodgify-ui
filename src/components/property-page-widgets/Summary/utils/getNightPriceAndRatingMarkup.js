import React, { Fragment } from 'react';
import { Segment } from 'semantic-ui-react';

import { getNightPriceMarkup } from 'utils/get-night-price-markup';
import { Rating } from 'elements/Rating';

/**
 * @param  {number} ratingNumber
 * @param  {string} nightPrice
 * @return {Object}
 */
export const getNightPriceAndRatingMarkup = (ratingNumber, nightPrice) => (
  <Fragment>
    <Segment>{getNightPriceMarkup(nightPrice, 'small')}</Segment>
    {!!ratingNumber && (
      <Segment className="is-rating-number">
        <Rating iconSize="tiny" ratingNumber={ratingNumber} />
      </Segment>
    )}
  </Fragment>
);
