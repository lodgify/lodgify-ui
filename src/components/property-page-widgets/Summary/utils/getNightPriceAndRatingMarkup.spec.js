import { mount } from 'enzyme';

import { getNightPriceAndRatingMarkup } from './getNightPriceAndRatingMarkup';

const props = {
  ratingNumber: 90,
  nightPrice: '280$',
};

const getMarkupAsRenderedComponent = (ratingNumber = props.ratingNumber) =>
  mount(getNightPriceAndRatingMarkup(ratingNumber, props.nightPrice));

describe('getNightPriceAndRatingMarkup', () => {
  it('should return the right children', () => {
    const wrapper = getMarkupAsRenderedComponent();

    expect(wrapper).toMatchSnapshot();
  });

  describe('the `ratingNumber`', () => {
    it('should not render the `Rating` if the `ratingNumber` is 0', () => {
      const wrapper = getMarkupAsRenderedComponent(0);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
