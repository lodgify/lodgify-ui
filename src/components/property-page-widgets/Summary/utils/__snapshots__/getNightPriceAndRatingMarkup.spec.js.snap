// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getNightPriceAndRatingMarkup if \`ratingNumber\` is \`0\` should return the right markup 1`] = `
<div>
  <Segment>
    <div
      className="ui segment"
    >
      <span>
        from 
        <Heading
          className={null}
          isColorInverted={false}
          size="small"
          textAlign={null}
        >
          <Header
            as="h4"
            className={null}
            inverted={false}
            textAlign={null}
          >
            <h4
              className="ui header"
            >
              280$
            </h4>
          </Header>
        </Heading>
        /night
      </span>
    </div>
  </Segment>
</div>
`;

exports[`getNightPriceAndRatingMarkup if \`ratingNumber\` is not \`0\` should return the right markup 1`] = `
<div>
  <Segment>
    <div
      className="ui segment"
    >
      <span>
        from 
        <Heading
          className={null}
          isColorInverted={false}
          size="small"
          textAlign={null}
        >
          <Header
            as="h4"
            className={null}
            inverted={false}
            textAlign={null}
          >
            <h4
              className="ui header"
            >
              280$
            </h4>
          </Header>
        </Heading>
        /night
      </span>
    </div>
  </Segment>
  <Segment
    className="is-rating"
  >
    <div
      className="ui segment is-rating"
    >
      <Rating
        iconSize="tiny"
        isShowingNumeral={true}
        ratingNumber={1}
      >
        1
        <Rating
          clearable="auto"
          disabled={true}
          maxRating={5}
          rating={1}
          size="tiny"
        >
          <div
            className="ui tiny disabled rating"
            onMouseLeave={[Function]}
            role="radiogroup"
          >
            <RatingIcon
              active={true}
              aria-checked={true}
              aria-posinset={1}
              aria-setsize={5}
              as="i"
              index={0}
              key="0"
              onClick={[Function]}
              onMouseEnter={[Function]}
              selected={false}
            >
              <i
                aria-checked={true}
                aria-posinset={1}
                aria-setsize={5}
                className="active icon"
                onClick={[Function]}
                onKeyUp={[Function]}
                onMouseEnter={[Function]}
                role="radio"
                tabIndex={0}
              />
            </RatingIcon>
            <RatingIcon
              active={false}
              aria-checked={false}
              aria-posinset={2}
              aria-setsize={5}
              as="i"
              index={1}
              key="1"
              onClick={[Function]}
              onMouseEnter={[Function]}
              selected={false}
            >
              <i
                aria-checked={false}
                aria-posinset={2}
                aria-setsize={5}
                className="icon"
                onClick={[Function]}
                onKeyUp={[Function]}
                onMouseEnter={[Function]}
                role="radio"
                tabIndex={0}
              />
            </RatingIcon>
            <RatingIcon
              active={false}
              aria-checked={false}
              aria-posinset={3}
              aria-setsize={5}
              as="i"
              index={2}
              key="2"
              onClick={[Function]}
              onMouseEnter={[Function]}
              selected={false}
            >
              <i
                aria-checked={false}
                aria-posinset={3}
                aria-setsize={5}
                className="icon"
                onClick={[Function]}
                onKeyUp={[Function]}
                onMouseEnter={[Function]}
                role="radio"
                tabIndex={0}
              />
            </RatingIcon>
            <RatingIcon
              active={false}
              aria-checked={false}
              aria-posinset={4}
              aria-setsize={5}
              as="i"
              index={3}
              key="3"
              onClick={[Function]}
              onMouseEnter={[Function]}
              selected={false}
            >
              <i
                aria-checked={false}
                aria-posinset={4}
                aria-setsize={5}
                className="icon"
                onClick={[Function]}
                onKeyUp={[Function]}
                onMouseEnter={[Function]}
                role="radio"
                tabIndex={0}
              />
            </RatingIcon>
            <RatingIcon
              active={false}
              aria-checked={false}
              aria-posinset={5}
              aria-setsize={5}
              as="i"
              index={4}
              key="4"
              onClick={[Function]}
              onMouseEnter={[Function]}
              selected={false}
            >
              <i
                aria-checked={false}
                aria-posinset={5}
                aria-setsize={5}
                className="icon"
                onClick={[Function]}
                onKeyUp={[Function]}
                onMouseEnter={[Function]}
                role="radio"
                tabIndex={0}
              />
            </RatingIcon>
          </div>
        </Rating>
      </Rating>
    </div>
  </Segment>
</div>
`;
